src/rdb.c

```diff
@@ -1315,17 +1315,18 @@ ssize_t rdbSaveDb(rio *rdb, int dbid, int rdbflags, long *key_counter) {
     written += res;
 
     /* Write the RESIZE DB opcode. */
-    uint64_t expires_size;
-    expires_size = dictSize(db->expires);
-    if ((res = rdbSaveType(rdb,RDB_OPCODE_RESIZEDB)) < 0) goto werr;
-    written += res;
-    if ((res = rdbSaveLen(rdb,db_size)) < 0) goto werr;
-    written += res;
-    if ((res = rdbSaveLen(rdb,expires_size)) < 0) goto werr;
-    written += res;
+    if (!server.cluster_enabled) {
+        unsigned long long expires_size = dbSize(db, EXPIRE_DICT);
+        if ((res = rdbSaveType(rdb,RDB_OPCODE_RESIZEDB)) < 0) goto werr;
```

Comment:
i think it's wrong not to save the RDB_OPCODE_RESIZEDB opcode when cluster mode is enabled.
the RDB should still contain this opcode, even if it's unlikely for the reader to use it.